src/amberscript/parser.cc

```diff
@@ -335,31 +335,58 @@ Result Parser::ParsePipelineAttach(Pipeline* pipeline) {
   if (!shader)
```

Comment:
This issue is not related to this PR, but as far as I checked, when `ATTACH` command is shown before its target shader, it reports "3: unknown shader in ATTACH command". Is this expected behavior?

See this example:
```
PIPELINE compute my_pipeline
  ATTACH my_shader TYPE compute ENTRY_POINT my_entry_point
END
SHADER multi my_shader GLSL
# shaders
END
```

Reply:
Do you think it is worth to add the following unit test in src/amberscript/parser_test.cc?

```
TEST_F(AmberScriptParserTest, PiplineMultiShaderAttach2) {
  std::string in = R"(
PIPELINE compute my_pipeline
  ATTACH my_shader TYPE compute ENTRY_POINT my_entry_point
END
SHADER multi my_shader GLSL
# shaders
END)";

  Parser parser;
  Result r = parser.Parse(in);
  ASSERT_TRUE(r.IsSuccess()) << r.Error();

  auto script = parser.GetScript();
  const auto& pipelines = script->GetPipelines();
  ASSERT_EQ(1U, pipelines.size());

  const auto* pipeline = pipelines[0].get();
  const auto& shaders = pipeline->GetShaders();
  ASSERT_EQ(1U, shaders.size());

  ASSERT_TRUE(shaders[0].GetShader() != nullptr);
  EXPECT_EQ(kShaderTypeMulti, shaders[0].GetShader()->GetType());
  EXPECT_EQ(kShaderTypeCompute, shaders[0].GetShaderType());
  EXPECT_EQ("my_entry_point", shaders[0].GetEntryPoint());
}
```